frappe/desk/query_report.py

```diff
@@ -216,20 +216,32 @@ def add_data_to_custom_columns(columns, result):
 	return data
 
 def reorder_data_for_custom_columns(custom_columns, columns, result, report_type):
-	custom_column_labels = [col["label"] for col in custom_columns]
+	if not result:
+		return []
 
 	if report_type == 'Query Report':
+		# Assume list result for query reports
+		# Query report columns exclisvuely use Label
+		custom_column_labels = [col["label"] for col in custom_columns]
 		original_column_labels = [col.split(":")[0] for col in columns]
+		return get_columns_from_list(custom_column_labels, original_column_labels, result)
+
+	if type(result[0]) == "list":
```

Comment:
`type` won't return a string, you'll have to do:

```suggestion
	if type(result[0]) == list:
```

Or even this works:

```suggestion
	if isinstance(result[0], list):
```

Reply:
`isinstance` ftw